src/JsonSchema/Uri/UriResolver.php

```diff
@@ -80,43 +82,45 @@ public function resolve($uri, $baseUri = null)
 
         $components = $this->parse($uri);
         $path = $components['path'];
-        
-        if (! empty($components['scheme'])) {
+
+        if (!empty($components['scheme'])) {
             return $uri;
         }
         $baseComponents = $this->parse($baseUri);
         $basePath = $baseComponents['path'];
-        
+
         $baseComponents['path'] = self::combineRelativePathWithBasePath($path, $basePath);
         if (isset($components['fragment'])) {
             $baseComponents['fragment'] = $components['fragment'];
         }
 
         return $this->generate($baseComponents);
     }
-    
+
     /**
      * Tries to glue a relative path onto an absolute one
      *
      * @param string $relativePath
      * @param string $basePath
-     * @return string Merged path
+     *
      * @throws UriResolverException
+     *
+     * @return string Merged path
      */
     public static function combineRelativePathWithBasePath($relativePath, $basePath)
     {
         $relativePath = self::normalizePath($relativePath);
         if ($relativePath == '') {
             return $basePath;
         }
-        if ($relativePath{0} == '/') {
+        if ($relativePath[0] == '/') {
```

Comment:
Is this intended to be `$relativePath[0]`, or `$relativePath->{'0'}`? Both options are a possibility; dynamic object fields are used elsewhere in the code. Is `$relativePath` always an array (or an object that implements array access), or can it sometimes be a non-array-compliant object?